pages/repo/raw.js

```diff
@@ -0,0 +1,244 @@
+import React, { useState, useEffect } from 'react'
+import Frame from '../../components/Frame'
+import RepoNav from '../../components/Repo/Nav'
+import initLocalStore from '../../lib/utils/localStorage'
+import { withRouter } from 'next/router'
+import withT from '../../lib/withT'
+import withAuthorization from '../../components/Auth/withAuthorization'
+import withMe from '../../lib/withMe'
+import { compose } from 'react-apollo'
+import { stringify, parse } from '@orbiting/remark-preset'
+import { css } from 'glamor'
+import {
+  A,
+  Button,
+  mediaQueries,
+  colors,
+  fontFamilies,
+  Checkbox,
+  linkRule
+} from '@project-r/styleguide'
+import { Router } from '../../lib/routes'
+import CircleIcon from 'react-icons/lib/md/lens'
+import InfoIcon from 'react-icons/lib/md/info-outline'
+import { Controlled as CodeMirror } from 'react-codemirror2'
+import {
+  UncommittedChanges,
+  withUncommitedChanges,
+  withUncommittedChangesMutation
+} from '../../components/VersionControl/UncommittedChanges'
+import BranchingNotice from '../../components/VersionControl/BranchingNotice'
+
+const styles = css({
+  background: colors.secondaryBg,
+  minHeight: 'calc(100vh - 80px)',
+  padding: 10,
+  '& .Checkbox': {
+    display: 'bloc',
+    margin: 5,
+    marginLeft: 'calc(50% + 210px)'
+  },
+  '& .Info': {
+    position: 'fixed',
+    right: 25,
+    top: 95
+  },
+  '& .CodeMirror': {
+    height: 'auto',
+    width: 800,
+    margin: 'auto',
+    border: `1px solid ${colors.divider}`,
+    fontFamily: fontFamilies.monospaceRegular,
+    fontSize: 14,
+    color: colors.text,
+    lineHeight: 2
+  },
+  '& pre.CodeMirror-line': {
+    padding: '0 15px'
+  }
+})
+
+// CodeMirror can only run in the browser
+if (process.browser) {
+  require('../../components/editor/utils/codemirror-md')
+  require('codemirror/addon/fold/foldcode')
+  require('codemirror/addon/fold/foldgutter')
+  require('codemirror/addon/fold/xml-fold')
+}
+
+const hasOpenSections = md => {
+  if (!md) return
+  // in synch with our remark-preset
+  // https://github.com/orbiting/mdast/blob/fabbd146dd16f7bdcbf7b699c5bac7161f14d197/packages/remark-preset/src/index.js#L23-L28
+  const openTags = md.match(/<section>\s*<h6>([^<]+)<\/h6>/g)
+  const closeTags = md.match(/<hr\s*\/>\s*<\/section>/g)
+  if (!openTags || !closeTags) return
+  return openTags.length !== closeTags.length
```

Comment:
```suggestion
  return openTags?.length !== closeTags?.length
```

because
```
> '</hr></section>'.match(/<section>\s*<h6>([^<]+)<\/h6>/g)
null
> '<hr/></section>'.match(/<hr\s*\/>\s*<\/section>/g)
["<hr/></section>"]
```

But in that case we still have an open section.